sections/delegate/Form.tsx

```diff
@@ -0,0 +1,303 @@
+		() =>
+			properDelegateAddress && address
+				? properDelegateAddress === ethers.utils.getAddress(address)
+				: null,
+		[properDelegateAddress, address]
+	);
+	const shortenedDelegateAddress = useMemo(() => truncateAddress(delegateAddress, 8, 6), [
+		delegateAddress,
+	]);
+
+	const getApproveTxData = useCallback(
+		(gas: Record<string, number>) => {
+			if (!(delegateApprovalsContract && properDelegateAddress && !delegateAddressIsSelf))
+				return null;
+			return [
+				delegateApprovalsContract,
+				APPROVE_CONTRACT_METHODS.get(action),
+				[properDelegateAddress, gas],
+			];
+		},
+		[delegateApprovalsContract, properDelegateAddress, action, delegateAddressIsSelf]
+	);
+
+	const onGoBack = () => router.back();
+	const onEnterAddress = (e: any) => setDelegateAddress((e.target.value ?? '').trim());
+	const onButtonClick = async () => {
+		if (!isWalletConnected) {
+			return connectWallet();
+		}
+		delegate();
+	};
+
+	const delegate = async () => {
+		setButtonState('delegating');
+		setTxModalOpen(true);
+		try {
+			const gas: Record<string, number> = {
+				gasPrice: getNormalizedGasPrice(gasPrice),
+				gasLimit: gasLimit!,
+			};
+			await tx(() => getApproveTxData(gas), {
+				showErrorNotification: (e: string) => setError(e),
+				showProgressNotification: (hash: string) =>
+					monitorTransaction({
+						txHash: hash,
+						onTxConfirmed: () => {},
+					}),
+				showSuccessNotification: (hash: string) => {},
+			});
+			setDelegateAddress('');
+			setAction(Action.APPROVE_ALL);
+		} catch {
+		} finally {
+			setButtonState(null);
+			setTxModalOpen(false);
+		}
+	};
+
+	// gas
+	useEffect(() => {
+		let isMounted = true;
+		(async () => {
+			try {
+				setError(null);
+				const data: any[] | null = getApproveTxData({});
+				if (!data) return;
+				const [contract, method, args] = data;
+				const gasEstimate = await getGasEstimateForTransaction(args, contract.estimateGas[method]);
+				if (isMounted) setGasLimitEstimate(getNormalizedGasLimit(Number(gasEstimate)));
+			} catch (error) {
+				// console.error(error);
+				if (isMounted) setGasLimitEstimate(null);
+			}
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [getApproveTxData]);
+
+	// already delegated
+	useEffect(() => {
+		if (!delegateApprovalsContract) return;
+		let isMounted = true;
+		(async () => {
+			if (!(properDelegateAddress && action)) return setAlreadyDelegated(false);
+			const alreadyDelegated = await delegateApprovalsContract[
+				GET_IS_APPROVED_CONTRACT_METHODS.get(action)!
+			](address, properDelegateAddress);
+			if (isMounted) setAlreadyDelegated(alreadyDelegated);
+		})();
+		return () => {
+			isMounted = false;
+		};
+	}, [delegateApprovalsContract, properDelegateAddress, address, action]);
+
+	return (
+		<div data-testid="form">
+			<FormContainer>
+				<Header>
+					<IconButton onClick={onGoBack}>
+						<Svg src={NavigationBack} />
+					</IconButton>
+				</Header>
+
+				<InputsContainer>
+					<AmountInput
```

Comment:
add `spellCheck={false}` to turn off spell checking

Reply:
Awesome.